storage/client/bucket_metadata.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/client/bucket_metadata.h"
+#include "storage/client/internal/nljson.h"
+#include "storage/client/internal/parse_rfc3339.h"
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+BucketMetadata BucketMetadata::ParseFromJson(std::string const& payload) {
+  auto json = storage::internal::nl::json::parse(payload);
+  BucketMetadata result{};
+  result.etag_ = json["etag"].get<std::string>();
+  result.id_ = json["id"].get<std::string>();
+  result.kind_ = json["kind"].get<std::string>();
```

Comment:
What if `"kind"` does not exist in the `json` object?

Reply:
How would this work for a hypothetical customer who wants to disable exceptions?